app/views/users/show.html.erb

```diff
@@ -1,4 +1,29 @@
-<h1>User Profile</h1>
-<p>Name: <%= @user.name %></p>
-<p>Photo: <%= @user.photo %></p>
-<p>Bio: <%= @user.bio %></p>
\ No newline at end of file
+<a href="<%= user_posts_path(@user) %>">
+  <button class="custom-back-btn">⬅</button>
+</a>
+<div class="custom-user-box">
+  <%= render partial: 'user_helper/user_photo', locals: { user: @user } %>
+  <%= render partial: 'user_helper/user_details', locals: { user: @user } %>
+</div>
+<div class="custom-bio">
+  <h3>Bio</h3>
+  <p>
+    <%= @user.bio %>
+  </p>
+</div>
+<% @user.recent_posts.each do |post| %>
+  <div class="custom-post custom-bio">
+    <h3><%= link_to "Post #{post.id}", post_path(post) %></h3>
+    <p><%= post.text %></p>
+    <div class='custom-post-interactions'>
+      <span>Comments: <%= post.comments_counter %>,</span>
+      <span>Likes: <%= post.likes_counter %></span>
+    </div>
+  </div>
+<% end %>
+<% end %>
+<div class="text-center">
+  <%= link_to user_posts_path(@user) do %>
+    <button class="custom-pagination">See all posts</button>
+  <% end %>
+</div>
```

Comment:
User#show is still not displaying as intended, as there appears to be multiple errors. Let's improve upon these aspects by ensuring:
- [ ] First, the error `ActionView::SyntaxErrorInTemplate in UsersController#show` is thrown. This is due to syntax error in the [app/views/users/show.html.erb](https://github.com/axcell-0/Blog-app/pull/6/files#diff-78ff736409d758722403cce31873ba803b3fb526d37398ca21caa557b54dfd95) on line 24. Let's correct that by removing `<% end %>` on _line 24_.

- [ ] Next, `NoMethodError` error resurfaces, as seen below:

![image](https://github.com/axcell-0/Blog-app/assets/12745474/0f48ee5c-7466-4bf8-a56e-560cccec35f0)

We can improve upon that aspect by first of improving the routes using nested routes like below:
```ruby
Rails.application.routes.draw do
  root 'users#index'

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show], controller: 'user_posts'
  end
end
```

And then let's ensure to use the correct path helpers accordingly in the [app/views/users/show.html.erb](https://github.com/axcell-0/Blog-app/pull/6/files#diff-78ff736409d758722403cce31873ba803b3fb526d37398ca21caa557b54dfd95) by changing line 16 to:

```ruby
<h3><%= link_to "Post #{post.id}", user_post_path(@user, post) %></h3>
```